scripts/apps/authoring/authoring/directives/AuthoringTopbarDirective.ts

```diff
@@ -31,7 +32,10 @@ export function AuthoringTopbarDirective(
         templateUrl: 'scripts/apps/authoring/views/authoring-topbar.html',
         link: function(scope) {
             function setActionsFromExtensions() {
-                scope.articleActionsFromExtensions = getArticleActionsFromExtensions(scope.item);
+                scope.articleActionsFromExtensions = partition(
```

Comment:
I would ask you to make this work with any number of groups, not only planning/non-planning. But since this is angular-authoring that we'll soon drop, it's fine to keep it.

What would be good to do which is easy is to not use a tuple here but 2 separate objects like `planningActions` / `nonPlanningActions`. Accessing the tuple using index is very confusing.

Reply:
I know, but given that it's in angular I don't think it's worth spending time on

Reply:
> What would be good to do which is easy is to not use a tuple here but 2 separate objects like planningActions / nonPlanningActions. Accessing the tuple using index is very confusing.

This one do anyway since it's quick. I don't want to mess up angular code even more if it's easy to avoid.